core/server/apps/amp/lib/helpers/amp_content.js

```diff
@@ -4,15 +4,43 @@
 // Turns content html into a safestring so that the user doesn't have to
 // escape it or tell handlebars to leave it alone with a triple-brace.
 //
-// Converts normal HTML into AMP HTML.
+// Converts normal HTML into AMP HTML with Amperize module and uses a cache to return it from
+// there if available. The cacheId is a combination of `updated_at` and the `slug`.
 
 var hbs             = require('express-hbs'),
-    ampContent;
+    Promise         = require('bluebird'),
+    Amperize        = require('amperize'),
+    moment          = require('moment'),
+    amperizeCache = {};
 
-ampContent = function () {
-    // var root = options.data.root;
-    // console.log('root:', root);
-    return new hbs.handlebars.SafeString(this.html);
-};
+function getAmperizeHTML(html, cacheId) {
+    if (!amperizeCache[cacheId]) {
+        return new Promise(function (resolve, reject) {
+            new Amperize().parse(html, function (err, res) {
+                if (err) {
+                    return reject(err);
+                }
+
+                amperizeCache[cacheId] = res;
+                return resolve(amperizeCache[cacheId]);
+            });
+        });
+    }
+
+    return Promise.resolve(amperizeCache[cacheId]);
+}
+
+function ampContent() {
+    var cacheId = moment(new Date(this.updated_at)) + this.slug,
```

Comment:
Using `post.id` instead of the `slug` sounds like a good idea to me. Still I need to use the `updated_at` in combination to create a unique cache id, because, the `updated_at` is the only reference, that the HTML that needs to be transformed has actually changed and we don't want to read it in that case.

In order to keep the cache clear of old and not valid entries, I will code three lines more and remove the outdated cache entry.